### PR TITLE
Rewrite of "Development" page in documentation

### LINKED ISSUE #367: Rewrite "Development" page in docs
Parts of the ["Development" page](http://lasagne.readthedocs.org/en/latest/user/development.html) are not exactly well-written, with unstructured notes thrown together. This should be rewritten to give a clear guide on what to contribute, and how to contribute, including some basic technical instructions (how to setup, how to test, how to compile docs, how to submit a PR).

Examples to draw from include:

* https://github.com/audreyr/cookiecutter/blob/master/CONTRIBUTING.rst / https://cookiecutter.readthedocs.org/en/latest/contributing.html (although that's a bit long)
* https://blocks.readthedocs.org/en/latest/development/index.html (although that could be structured a bit better)
* http://scikit-learn.org/stable/developers/ (although again, that could be structured better)

Hmm, I don't find a really perfect example, but at least those examples are much more complete in terms of content than ours.

### PR DESCRIPTION
**Pull request by f0k**
Resolves #367.

Mentions core team by name, moves the design goals further to the front, gives suggestions on what to contribute and technical instructions on how to contribute.

### CODE REVIEW

File: docs/user/development.rst [hunk 1]
```diff
@@ -2,151 +2,233 @@ Development
...
+through the implementation.
 
-You should also `install pylearn2 <http://deeplearning.net/software/pylearn2/#download-and-installation>`_
-to prevent warnings while generating the documentation.
 
-The documentation can be built with ``make html``.
+Write documentation
+~~~~~~~~~~~~~~~~~~~
 
-The documentation is written in numpydoc syntax. Information about numpydoc
```
Comment (craffel) [inline docs/user/development.rst]: We shouldn't omit reference to numpydoc completely, right?

**Comment by craffel** [comment]
LG(reat)TM in general.

File: docs/user/development.rst (hunk 1 above)
Comment (f0k) [inline docs/user/development.rst]: We're linking to this one instead: https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt
Should be fine as that's the one you need to obey, shouldn't it?

Comment (craffel) [inline docs/user/development.rst]: Ah, I see it there.  Looks good.

**Comment by benanne** [comment]
Seconded!

**Comment by skaae** [comment]
Looks very good

**Comment by f0k** [comment]
Merged! Thanks for checking it!

**Pull request merged**